src/proxy-builder.test.ts

```diff
@@ -12,22 +12,30 @@ describe("Test configureProxy", () => {
     process.env.HTTPS_PROXY = https_env;
     const result = configureProxy("https://test.com:8000");
 
-    expect(result).toEqual({
-      proxy: {
-        hostname: "testproxy.com",
-        port: 8000,
-        protocol: "https:",
-        auth: {
-          username: "",
-          password: "",
-        },
+    expect(result.proxy).toEqual({
+      hostname: "testproxy.com",
+      port: 8000,
+      protocol: "https:",
+      auth: {
+        username: "",
+        password: "",
       },
-    } as RequestProxy);
+    });
+    expect(result.httpsAgent).toBeUndefined();
+  });
+
+  test("with tunnel agent", () => {
+    process.env.HTTP_PROXY = http_env;
```

Comment:
we need to test https over http proxy

process.env.HTTPS_PROXY = http_env;

Reply:
That seems like a bad practice? Wouldn't you just write your http proxy to http_proxy? We can of course write a case for this.

Reply:
that's what fix is for.
ideally you need to add fallback to http incase of https proxy
https://github.com/request/request/blob/3c0cddc7c8eb60b470e9519da85896ed7ee0081e/lib/getProxyFromURI.js#L69

but here there is no fallback
https://github.com/farawaysouthwest/axios-proxy-builder/blob/master/src/utils.ts#L61

hence, i had to set 
process.env.HTTPS_PROXY = http_env;

Reply:
if you add the fallback logic in this code.,
then process.env.HTTP_PROXY = http_env; this will do the job and test

Reply:
i have added here https://github.com/farawaysouthwest/axios-proxy-builder/pull/4/commits/6537175f0ec4014b72b858801b3d48a4ba7dc8b4

Reply:
Ok, I believe everything is in place?

Reply:
You're too fast for me man!😄

Reply:
haha. closed that PR. merge conflict. messed up few things